internal/provider/org/role_resource.go

```diff
@@ -0,0 +1,425 @@
+		resp.Diagnostics.Append(y...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+
+	// Set refreshed state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *orgRoleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var (
+		state *orgRoleResourceModel
+		err   error
+		role  *govcd.Role
+	)
+
+	// Read state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Org and role deletion is accessible only in administrator
+	adminOrg, err := r.client.Vmware.GetAdminOrgByNameOrId(r.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("[role update] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get the role
+	if state.ID.IsNull() {
+		role, err = adminOrg.GetRoleByName(state.Name.ValueString())
+	} else {
+		role, err = adminOrg.GetRoleById(state.ID.ValueString())
+	}
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			tflog.Info(ctx, "[DEBUG] Unable to find role. Removing from tfstate")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("[role delete] Error retrieving role", err.Error())
+		return
+	}
+	err = role.Delete()
+	if err != nil {
+		resp.Diagnostics.AddError("[role delete] Error deleting role", err.Error())
+		return
+	}
+}
+
+func (r *orgRoleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var (
+		plan, state *orgRoleResourceModel
+		err         error
+		role        *govcd.Role
+	)
+
+	// Get state and plan
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get Org and role update is accessible only in administrator
+	adminOrg, err := r.client.Vmware.GetAdminOrgByNameOrId(r.client.GetOrg())
+	if err != nil {
+		resp.Diagnostics.AddError("[role update] Error retrieving Org", err.Error())
+		return
+	}
+
+	// Get the role
+	role, err = adminOrg.GetRoleById(state.ID.ValueString())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			tflog.Info(ctx, "[DEBUG] Unable to find role. Removing from tfstate")
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("[role update] Error retrieving role", err.Error())
+		return
+	}
+
+	// Update the role Name or Description
+	if (plan.Name.Equal(state.Name)) || (plan.Description.Equal(state.Description)) {
+		role.Role.Name = plan.Name.ValueString()
+		role.Role.Description = plan.Description.ValueString()
+		_, err = role.Update()
+		if err != nil {
+			resp.Diagnostics.AddError("[role update] Error updating role", err.Error())
+			return
+		}
+	}
+
+	// Check rights are valid
+	rights := make([]govcdtypes.OpenApiReference, 0)
+	for _, right := range plan.Rights.Elements() {
+		rg := strings.Trim(right.String(), "\"")
+		x, err := adminOrg.GetRightByName(rg)
```

Comment:
`rg` assignement is unnecessary 

```suggestion
	x, err := adminOrg.GetRightByName(strings.Trim(right.String(), "\""))
```